### PR TITLE
feat: update to manifest v3 (Resolves: #32, #33, #29)

### PR DESCRIPTION
**Pull request by cindyli**
Resolves: #32
Resolves: #33
Resolves: #29 

Migration docs:
- [Manifest V3 transition timeline](https://developer.chrome.com/blog/mv2-transition/)
- [Migrating to Manifest V3](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,34 +26,32 @@
...
+        "test": "run-s test:node test:browser",
+        "test:node": "nyc node tests/node/all-tests.js",
+        "test:browser": "testem ci --file tests/browser/testem.js",
+        "test:report": "nyc report -r text-summary -r html -r json",
+        "posttest": "npm run test:report",
+        "clean": "run-p clean:*",
+        "clean:build": "rimraf dist",
+        "clean:test": "rimraf reports coverage",
+        "build": "run-s -l clean:build build:*",
+        "build:dist": "cpy . ../dist --cwd=src; cpy . ../../dist/lib/infusion --cwd=node_modules/infusion"
```
Comment (amb26) [inline package.json:40]: I think the ; operator is non-portable - I get an error running this line on Windows, replacing it with & allows the build to complete

File: src/js/enactors.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        }
+    },
+    markup: {
+        tocContainer: "<div class=\"flc-toc-tocContainer uioPlus-toc-tocContainer\"></div>"
+    }
+});
+
+uioPlus.enactor.tableOfContents.injectToCContainer = function (that) {
+    if (!that.locate("tocContainer").length) {
+        if (that.content.length === 1) {
```
Comment (amb26) [inline src/js/enactors.js:286]: A comment explaining why the variant strategy is necessary here would be helpful

File: package.json [hunk 3]
```diff
@@ -26,34 +26,32 @@
...
-        "grunt-cli": "1.4.3",
-        "grunt-contrib-clean": "2.0.0",
-        "grunt-contrib-copy": "1.0.0",
-        "grunt-contrib-stylus": "1.2.0",
-        "grunt-shell": "4.0.0",
         "node-jqunit": "1.1.9",
+        "npm-run-all": "4.1.5",
         "nyc": "15.1.0",
         "rimraf": "3.0.2",
-        "sinon": "13.0.1",
```
Comment (jobara) [inline package.json:55]: I'm getting an error when trying to run the tests that it can't find sinon.

Comment (cindyli) [reply package.json:55]: Interesting. `sinon-chrome` pulls in `sinon` on my computer. I've added the `sinon`. Can you try again? Thanks.

**Comment by jobara** [comment]
With UIO+ enabled the browser console reports the following error:

```error
{
    "message": "Assertion failure - check console for more details: No container element was found for selector .uioPlus",
    "stack": "Error: Assertion failure - check console for more details: No container element was found for selector .uioPlus\n    at new fluid.FluidError (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:2470)\n    at fluid.builtinFail (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:3038)\n    at fluid.event.firer.fire (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:25883)\n    at fluid.fail (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:3289)\n    at fluid.container (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:224673)\n    at fluid.containerForViewComponent (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:228075)\n    at fluid.expander.invokeFunc [as typeFunc] (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:116253)\n    at fluid.expandImmediateImpl (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:113497)\n    at fluid.expandImmediate (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:113252)\n    at fluid.memberFromRecord (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:26907)"
}
```

**Comment by jobara** [comment]
On engadget.com when enabling Text-to-Speech I get the following error.

```error
Error in event handler: Error: Assertion failure - check console for more details: More than one (32) container elements were found for selector article, [role~='article'], .article, #article with selector name article in context null
    at new fluid.FluidError (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:2470)
    at fluid.builtinFail (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:3038)
    at fluid.event.firer.fire (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:25883)
    at fluid.fail (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:128:3289)
    at fluid.container (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:224673)
    at fluid.containerForViewComponent (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:228075)
    at fluid.expander.invokeFunc [as typeFunc] (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:116253)
    at fluid.expandImmediateImpl (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:113497)
    at fluid.expandImmediate (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:113252)
    at fluid.memberFromRecord (chrome-extension://gbenfaceipnpnobakclnelebaeinlelm/lib/infusion/dist/infusion-uio.js:137:26907)
```

File: src/js/enactors.js [hunk 4]
```diff
@@ -308,8 +308,8 @@ fluid.defaults("uioPlus.enactor.selfVoicing", {
         controllerParentContainer: ".flc-prefs-selfVoicingWidget",
         domReaderContent: ".flc-orator-content"
     },
-    domReaderContent: ["domReaderContent", "article", "main", "genericContent"],
-    controllerParentContainer: ["controllerParentContainer", "article", "main", "genericContent"],
+    domReaderContent: ["domReaderContent", "main", "article"],
+    controllerParentContainer: ["controllerParentContainer", "main", "article"],
```
Comment (jobara) [inline src/js/enactors.js]: I think we still need the `"genericContent"` option here which is likely defined in a parent grade. It provides more fallbacks to find the page content.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
-        node-version: [14.x, 16.x]
+        node-version: [16.x]
```
Comment (jobara) [inline .github/workflows/ci.yml:14]: Might as well add Node 18 now too.

```suggestion
        node-version: [16.x, 18.x]
```

File: README.md [hunk 6]
```diff
@@ -110,7 +99,7 @@ _**NOTE:** Published versions can be installed from the [Chrome Web Store](
 
 ### BSD 3-Clause
 
-* [Infusion v3.0.0-dev](https://fluidproject.org/infusion.html)
+* [Infusion v4.0.0](https://fluidproject.org/infusion.html)
```
Comment (jobara) [inline README.md:102]:
```suggestion
* [Infusion v4.6.0](https://fluidproject.org/infusion.html)
```

File: src/js/enactors.js (hunk 4 above)
Comment (cindyli) [reply src/js/enactors.js]: You are probably looking at an old version of this file. `domReaderContent` and `controllerParentContainer` are used by subcomponents `domReader` and `controller`. As these sub-components are now disabled for turning off the page level reading. These lines have been removed in the last commit.

**Review by jobara — APPROVED** [review]

**Pull request merged**